charts/console/chart_test.go

```diff
@@ -251,3 +257,64 @@ func TestAppVersion(t *testing.T) {
 
 	require.Equalf(t, expected, actual, "Chart.yaml's version should be %q; got %q\nDid you forget to update Chart.yaml before minting a release?\nMake sure to bump appVersion as well!", expected, actual)
 }
+
+func TestIntegrationChart(t *testing.T) {
+	testutil.SkipIfNotIntegration(t)
+
+	log := zaptest.NewLogger(t)
+	w := &zapio.Writer{Log: log, Level: zapcore.InfoLevel}
+	wErr := &zapio.Writer{Log: log, Level: zapcore.ErrorLevel}
+
+	consoleChart := "."
+
+	h := helmtest.Setup(t)
+
+	t.Run("basic test", func(t *testing.T) {
+		ctx := testutil.Context(t)
+
+		env := h.Namespaced(t)
+
+		r, err := rand.Int(rand.Reader, new(big.Int).SetInt64(1799999999))
+		require.NoError(t, err)
+
+		partial := PartialValues{
+			Image: &PartialImage{
+				Repository: ptr.To("redpandadata/console-unstable"),
+				Tag:        ptr.To("master-2bb3af4"),
+			},
+			Config: map[string]any{
+				"kafka": map[string]any{
+					"brokers": []string{"redpanda-0.redpanda.default.svc.cluster.local:9092"},
+				},
+			},
+		}
+
+		chartReleaseName := fmt.Sprintf("chart-%d", r.Int64())
+		consoleRelease := env.Install(ctx, consoleChart, helm.InstallOptions{
+			Values:    partial,
+			Name:      chartReleaseName,
+			Namespace: env.Namespace(),
+		})
+
+		client := Client{Ctl: env.Ctl(), Release: &consoleRelease}
+
+		dot := &helmette.Dot{
+			Values:  *helmette.UnmarshalInto[*helmette.Values](partial),
+			Release: helmette.Release{Name: consoleRelease.Name, Namespace: consoleRelease.Namespace},
+			Chart: helmette.Chart{
+				Name: "console",
+			},
+		}
+
+		cleanup, err := client.ExposeConsole(ctx, dot, w, wErr)
```

Comment:
Oh there's no need for all this. `kube.Dialer` will let you tunnel directly into a Pod by it's name. There are examples of how to put it into an `http.Client` in the tests for it. Sorry you had to copy all of the client code 😓